Poisson_surface_reconstruction_3/include/CGAL/Poisson_mesh_domain_3.h

```diff
@@ -0,0 +1,411 @@
+// Copyright (c) 2025 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Sven Oesau
+
+#ifndef CGAL_POISSON_MESH_DOMAIN_3_H
+#define CGAL_POISSON_MESH_DOMAIN_3_H
+
+#include <CGAL/license/Poisson_surface_reconstruction_3.h>
+
+#include <CGAL/Labeled_mesh_domain_3.h>
+#include <CGAL/Poisson_reconstruction_function.h>
+
+namespace CGAL {
+
+/*!
+\ingroup PkgPoissonSurfaceReconstruction3Ref
+
+\brief The class `Poisson_mesh_domain_3` derives from `Labeled_mesh_domain_3` for the handling of `Poisson_reconstruction_function`.
+
+ This class has a constructor taking a labeling function. It has also a static template member
+ function that acts as named constructor:
+ <ul><li>`create_Poisson_mesh_domain()`</li>, to create a domain from a `Poisson_reconstruction_function`</ul>
+
+\tparam BGT is a geometric traits class that provides
+    the basic operations to implement intersection tests and intersection computations through a bisection
+    method.This parameter must be instantiated with a model of the concept `BisectionGeometricTraits_3`.
+
+\cgalModels{MeshDomain_3}
+
+\sa `CGAL::Labeled_mesh_domain_3`
+\sa `CGAL::make_mesh_3()`
+*/
+template<class BGT>
+class Poisson_mesh_domain_3
+    : public Labeled_mesh_domain_3<BGT>
```

Comment:
the doc has a lot of noise about creating gray level images, and other public inherited types and functions. Do we need the public inheritance?

Reply:
There are a bunch of member functions, e.g., creation of the functor structs and some methods for surface patches and subdomain indices, that are required and publicly inherited. I can make the inheritance protected and just recreate those methods (that may call the protected hidden ones).

The Poisson_mesh_domain_3.h would be bigger, but some of those confusing creation methods would be gone.